pkg/util/util.go

```diff
@@ -177,6 +178,30 @@ func CompareVersions(left string, right string) (int, error) {
 	return leftVersion.Compare(rightVersion), nil
 }
 
+func BreakingChangeAllowed(left string, right string) (bool, []error, error) {
```

Comment:
I'd suggest to only return `bool` and `error`. Let's add https://github.com/hashicorp/go-multierror and wrap multiple errors.